### PR TITLE
make_recipe function and tests

### LINKED ISSUE #25: write make_recipe test

### LINKED ISSUE #26: write make_recipe function

### LINKED ISSUE #36: Code review: Anny review make_recipe (R) (Sam)
Similar to milestone1, we expect you to follow GitHub Flow workflow. Additionally, each team member must do at least one code review and each member must have some part of their code reviewed by other team members. So please make sure that one of the Python or R functions (and its corresponding unit test) you are responsible for is ready to be reviewed by Wednesday, March 4th, end of the day.

### PR DESCRIPTION
**Pull request by SamEdwardes**
Closes #25 (tests), closes #26 (write function), and closes #36 (code review).

### CODE REVIEW

File: R/make_recipe.R [hunk 1]
```diff
@@ -3,26 +3,98 @@
...
 #' The `make_recipe()` function is used to quickly apply common data preprocessing techniques
 #'
-#' @param data A dataframe containing training data, validation data, X, and y
-#' @param recipe A string specifying which recipe to apply to the data
-#' @param create_train_test Boolean, if TRUE will partition data into train and test
-#' @param create_train_valid_test Boolean, if TRUE will partition data into train, valid, and test
+#' @param X A dataframe containing training data, validation data, and testing data (should contain X and y).
+#' @param y The name of the response column (as a string, e.g. "response_variable").
+#' @param recipe A string specifying which recipe to apply to the data. See "The recipe parameter" section below for details.
+#' @param splits_to_return A string speciying how to split the data. "train_test" to return train and test splits, "train_test_valid" to return train, test, and validation data, "train" to return all data without splits.
```
Comment (annychih) [inline R/make_recipe.R]: little typo:

```suggestion
#' @param splits_to_return A string specifying how to split the data. "train_test" to return train and test splits, "train_test_valid" to return train, test, and validation data, "train" to return all data without splits.
```

File: R/make_recipe.R [hunk 2]
```diff
@@ -3,26 +3,98 @@
...
+#' @param random_seed An integer. The random seed to set for splitting data to create reproducible results. By default NULL
+#' @param train_valid_prop A float. The proportion to split the data by. Should range between 0 to 1. By default = 0.8
 #'
-#' @return A list of dataframes e.g. list(train, valid, test)
-#'
-#' @section Recipes
+#' @return A list of dataframes e.g. list(X_train, X_valid, X_test, y_train, y_valid, y_test)
 #'
+#' @section The recipe parameter:
 #' The following recipes are available currently available to pass into the `recipe` parameter:
```
Comment (annychih) [inline R/make_recipe.R]:
```suggestion
#' The following recipes are available currently to pass into the `recipe` parameter:
```

File: R/make_recipe.R [hunk 3]
```diff
@@ -3,26 +3,98 @@
...
+#' * "ohe_and_standard_scaler" - Apply one hot encoding to categorical features and standard scaler to numeric features
 #'
-#' "ohe_and_standard_scaler"
+#' More recipes are under development and will be released in future updates.
 #'
-#' Apply one hot encoding to categorical features and standard scaler to numeric features
+#' @md
```
Comment (annychih) [inline R/make_recipe.R:21]: Was this added intentionally?

```suggestion
#'
```

**Review by annychih — CHANGES_REQUESTED** [review]
I'm not sure if we're supposed to put the `testthat()` tests within our functions or within the .R files within the `testthat` folder. Assuming that the `testthat()` tests within the `make_recipe()` function are where they belong and the function passed your "black box" tests, this pull request looks pretty good!

File: R/make_recipe.R (hunk 3 above)
Comment (SamEdwardes) [reply R/make_recipe.R:21]: Yes this was added intentionally. It is telling R that when building the documentation to treat my input as a markdown. It allows for much easier format when writing doc strings. So this must be kept.

File: R/make_recipe.R (hunk 1 above)
Comment (SamEdwardes) [reply R/make_recipe.R]: good catch!

Comment (SamEdwardes) [reply R/make_recipe.R]: good catch

File: R/make_recipe.R (hunk 2 above)
Comment (SamEdwardes) [reply R/make_recipe.R]: good catch

**Comment by SamEdwardes** [comment]
> I'm not sure if we're supposed to put the `testthat()` tests within our functions or within the .R files within the `testthat` folder. Assuming that the `testthat()` tests within the `make_recipe()` function are where they belong and the function passed your "black box" tests, this pull request looks pretty good!

Thanks for the feedback Anny. I corrected the typos and left a comment about the `@md` tag.

In regards to `testthat::` in make_recipe, I am not using it as a unit test or anything like that. I am just validating the input from the users. There are many ways you could go about this, I just like how testthat is easy to read. I don't think there is any problem with it existing in the function. Let me know if you agree! If so, it should be ready to merge.

**Review by annychih — APPROVED** [review]

**Comment by annychih** [comment]
Thanks for the changes and notes!

**Pull request merged**